tests/test_cms_menus.py

```diff
@@ -190,3 +246,187 @@ def test_get_roots_with_versioning_disabled(self):
+                a1 = getattr(n1, attr)
+                a2 = getattr(n2, attr)
+                msg = '%r != %r with %r, %r (%s)' % (a1, a2, n1, n2, attr)
+                self.assertEqual(a1, a2, msg)
+            self.assertTreeQuality(n1.children, n2.children)
+
+    def test_menu_without_softroots(self):
+        """
+        Tree in fixture :
+               root
+                   aaa
+                       aaa1
+                           ccc
+                               ddd
+                       aaa2
+                   bbb
+        tag: show_menu 0 100 0 100
+        expected result 1:
+               0:root
+                  1:aaa
+                     2:aaa1
+                        3:ccc
+                           4:ddd
+                     5:aaa2
+                  6:bbb
+        """
+        page_url = self.aaa_pagecontent.page.get_absolute_url()
+        context = self.get_context(page_url, page=self.root_pagecontent.page)
+        # response = Context(self.client.get(page_url).rendered_content)
+        # context.rendered_comtent has the page data , which need to be rendered to a template
+        tpl = Template("{% load menu_tags %}{% show_menu 0 100 100 100 %}")
+        tpl.render(context)
+        hard_root = context['children']
+        mock_tree = [
+            AttributeObject(title=self.root.title, level=0, children=[
+                AttributeObject(title=self.aaa.title, level=1, children=[
+                    AttributeObject(title=self.aaa1.title, level=2, children=[
+                        AttributeObject(title=self.ccc.title, level=3, children=[
+                            AttributeObject(title=self.ddd.title, level=4, children=[])
+                        ])
+                    ]),
+                    AttributeObject(title=self.aaa2.title, level=2, children=[])
+                ]),
+                AttributeObject(title=self.bbb.title, level=1, children=[])
+            ])
+        ]
+        self.assertTreeQuality(hard_root, mock_tree)
+
+    def test_menu_with_softroot(self):
+        """
+        Tree in fixture :
+               root (soft_root)
+                   aaa
+                       aaa1
+                           ccc
+                               ddd
+                       aaa2
+                   bbb
+        tag: show_menu 0 100 0 100
+        expected result 1:
+               0:root
+                  1:aaa
+                     2:aaa1
+                        3:ccc
+                           4:ddd
+                     5:aaa2
+                  6:bbb
+        """
+        self.root.soft_root = True
+        page_url = self.aaa_pagecontent.page.get_absolute_url()
+        context = self.get_context(page_url, page=self.root_pagecontent.page)
+        tpl = Template("{% load menu_tags %}{% show_menu 0 100 100 100 %}")
+        tpl.render(context)
+        soft_root = context['children']
+        mock_tree = [
+            AttributeObject(title=self.root.title, level=0, children=[
```

Comment:
This test does not work for softroots, it's exactly the same as the test above it. I don't think that the original test in the cms is anygood either. Why would you test for a softroot on the root, it's already the root!

Try this, Make self.aaa the soft_root, and then use the page_url as 
```
page = self.aaa1_pagecontent.page
context = self.get_context(page.get_absolute_url(), page=page)
```

Reply:
Setting self.navnode.soft_root = True is always seen as False in is_selected. You will need to copy all of the nodes from the setup and create them here.

Reply:
Updated the tests with correct page request and expected results.